### PR TITLE
To Do list: add & remove #2

### PR DESCRIPTION
**Pull request by Nejjari001**
Hi,
 **In this particular project To-Do List: Add & Remove :**

- Remove all hardcoded items from the tasks array.

- Create a new JavaScript file for the new functionality.

- Implement a function for adding a new task (add a new element to the array).

- Implement a function for deleting a task (remove an element from the array).

- Implement a function for editing task descriptions.

- By default, new tasks have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).

- Deleting a task update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) is set to 1).

- All changes to the To-Do List are saved in local storage.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -25,8 +25,6 @@ <h3>Today's To Do.</h3>
       </div>
       <div class="display"></div>
     </div>
-    <!-- <div class="display"></div>
```
Comment (BenMukebo) [inline src/index.html:28]: - [ ] Great job 👏🏻 However I couldn't open your app in the browser even after running `npm install`, `npm run build`, 'npm run start`, or `npm run dev`. instead, I get this error message 👇🏻 as shown in the screenshot. Kindly Make sure to fix this error 😉 👍🏻 


Screenshot from 2022-10-06 13-14-15

- I also test this with another computer and it's the same result 👇🏻 

image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @Nejjari001,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- No linter errors. ✔️ 
- GitHub flow was correctly used.  ✔️    
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [ ] When I click the link to the `MIT.md` file that you added, I get this 404 `error`:

00Mit

The reason for this is that there is no `MIT.md` file in the root directory of this project. It will be nice to add and update this section(`License`) with the type of License and a link to the license file, as this will make your readme more informative and professionalKindly attach the file. 👍


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Nejjari001, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No linter errors ✔️
- Correct Github flow ✔️
- Descriptive PR ✔️
- Professional README file ✔️


 ## Required Changes ♻️
_Check the comments under the review._

- [ ] As the last code reviewer said I couldn't open your project in my browser either. 👇🏽 I recommend trying to record a video.

image


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Nejjari001** [comment]
Hi @franclobo,
This is my recorded video of the live project it works perfectly on my side kindly double-check it 👍 
https://www.loom.com/share/29e2a1fe7efb4315884028f7cc4a9b17

File: src/index.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ const totalTasks = [
   },
   {
```
Comment (KDlamini) [inline src/index.js:21]: [OPTIONAL] Sometimes when I add tasks they are added twice and sometimes when I delete a task it is not deleted in the localStorage, although it disappears on the page.

- Kindly look into it to make sure that it does not cause any inconsistencies with the functionality of your to-do app.

**Review by KDlamini — APPROVED** [review]
### Hi @Nejjari001 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Great PR title and description
✔️ Application runs smoothly
✔️ No linter errors

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (Nejjari001) [reply src/index.js:21]: Hi @KDlamini,
Thank you for your kind and fast review, yes I will double-check it 👍

**Pull request merged**